applications/mlflow/Taskfile.yml

```diff
@@ -0,0 +1,804 @@
+          echo "This may indicate issues with the service or port forwarding."
+          echo "Port forwarding log:"
+          cat $PORT_FORWARD_LOG
+          echo "Pod logs:"
+          kubectl logs -n {{.NAMESPACE}} -l app.kubernetes.io/name=mlflow --tail=20 || true
+          echo "Continuing anyway, but tests may fail."
+        fi
+        
+        echo "Port forwarding setup completed."
+
+  # Local installation task (renamed from test:install:local)
+  install:helm:local:
+    desc: Install MLflow with local Helm charts for development (with Replicated SDK disabled)
+    deps: [add:repos:helm, setup:namespaces]
+    cmds:
+      - echo "Installing MLflow with local Helm charts (Replicated SDK disabled)..."
+      - |
+        # Prepare values arguments if provided
+        MLFLOW_VALUES_ARGS=""
+        if [ -n "$MLFLOW_VALUES" ]; then
+          echo "Using MLflow values file: $MLFLOW_VALUES"
+          # Check if values file exists
+          if [ ! -f "$MLFLOW_VALUES" ]; then
+            echo "ERROR: Values file '$MLFLOW_VALUES' does not exist"
+            exit 1
+          fi
+          MLFLOW_VALUES_ARGS="--values $MLFLOW_VALUES"
+          echo "Values args: $MLFLOW_VALUES_ARGS"
+        else
+          echo "No custom values file provided. Using default values."
+        fi
+        
+        # Install infra chart from local directory
+        echo "Installing infra chart from local directory..."
+        helm upgrade --install infra {{.CHART_DIR}}/infra \
+          --namespace {{.NAMESPACE}} \
+          --wait --timeout {{.TIMEOUT}} --debug || { 
+          echo "ERROR: Failed to install infra chart from {{.CHART_DIR}}/infra"
+          exit 1
+        }
+        
+        # Install MLflow chart from local directory with custom values
+        # Note: We explicitly disable the Replicated SDK for local development to avoid
+        # dependencies on the Replicated platform during development
+        echo "Installing mlflow chart from local directory (Replicated SDK disabled)..."
+        echo "Using values args: $MLFLOW_VALUES_ARGS"
+        helm upgrade --install mlflow {{.CHART_DIR}}/mlflow \
+          --namespace {{.NAMESPACE}} \
+          --set replicated.enabled=false \
+          $MLFLOW_VALUES_ARGS \
+          --wait --timeout {{.TIMEOUT}} --debug || { 
+          echo "ERROR: Failed to install mlflow chart from {{.CHART_DIR}}/mlflow"
+          exit 1
+        }
+        
+        echo "Local Helm installation completed successfully."
+      - task: forward:port
+
+  # App test task
+  run:tests:app:
+    desc: Run application tests against the running MLflow service
+    cmds:
+      - echo "Running application tests against MLflow on localhost:{{.PORT}}..."
+      - |
+        echo "Installing Python dependencies for tests..."
+        pip3 install setuptools mlflow==2.11.0 pandas>=2.0.0 scikit-learn>=1.3.0 requests>=2.31.0 urllib3>=2.0.0
```

Comment:
Ouch this doesn't look like how I would want to manage my test dependencies :)

pyproject.toml or at least a requirements file that get's installed would be a lot easier to maintain and update. There is tooling that can process those where as this is going to always require manual updates.

Reply:
yeah, for sure

Reply:
I'm fine with test tasks being follow on from here, maybe leave a note in a comment to make that clear to future travelers.